File: frontend/couple_map/lib/models/memory/media_file.dart
enum MediaFileType {
  IMAGE,
  VIDEO,
  AUDIO,
}

class MediaFile {
  final int mediaFileId;
  final String fileUrl;
  final String originalFilename;
  final MediaFileType mediaFileType;
  final int fileSize;
  final int displayOrder;

  MediaFile({
    required this.mediaFileId,
    required this.fileUrl,
    required this.originalFilename,
    required this.mediaFileType,
    required this.fileSize,
    required this.displayOrder,
  });

  factory MediaFile.fromJson(Map<String, dynamic> json) {
    return MediaFile(
      mediaFileId: json['mediaFileId'] as int,
      fileUrl: json['fileUrl'] as String,
      originalFilename: json['originalFilename'] as String,
      mediaFileType: MediaFileType.values.firstWhere(
        (e) => e.name == json['mediaFileType'],
        orElse: () => MediaFileType.IMAGE,
      ),
      fileSize: json['fileSize'] as int,
      displayOrder: json['displayOrder'] as int,
    );
  }
}

File: backend/src/main/java/com/couplemap/mediafile/domain/MediaFile.java
package com.couplemap.mediafile.domain;

import com.couplemap.global.common.BaseEntity;
import com.couplemap.global.s3.S3UploadDto;
import com.couplemap.memory.domain.Memory;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "media_files")
public class MediaFile extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "media_file_id")
    private Long mediaFileId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "memory_id", nullable = false)
    private Memory memory;

    // 파일 접근용
    @Column(name = "file_url", nullable = false, length = 500)
    private String fileUrl;

    // 파일 삭제용
    @Column(name = "file_key", nullable = false, length = 300)
    private String fileKey;

    @Column(name = "original_filename", length = 255)
    private String originalFilename;

    @Enumerated(EnumType.STRING)
    @Column(name = "media_file_type", nullable = false)
    private MediaFileType mediaFileType;  // IMAGE, VIDEO , AUDIO

    @Column(name = "file_size")
    private Long fileSize;

    @Column(name = "display_order")
    private Integer displayOrder;

    @Builder
    private MediaFile(Memory memory, String fileUrl, String fileKey, String originalFilename, MediaFileType mediaFileType, Long fileSize, Integer displayOrder) {
        this.memory = memory;
        this.fileUrl = fileUrl;
        this.fileKey = fileKey;
        this.originalFilename = originalFilename;
        this.mediaFileType = mediaFileType;
        this.fileSize = fileSize;
        this.displayOrder = displayOrder;
    }

    public static MediaFile from(Memory memory, S3UploadDto s3Dto, MultipartFile file, MediaFileType type, int order) {
        return MediaFile.builder()
                .memory(memory)
                .fileUrl(s3Dto.getUrl())
                .fileKey(s3Dto.getKey())
                .originalFilename(file.getOriginalFilename())
                .mediaFileType(type)
                .fileSize(file.getSize())
                .displayOrder(order)
                .build();
    }
}
